### ISSUE TITLE
Brightness controls not working

### ISSUE THREAD

**Issue by jball-98** [nguyenhung9x2018/HP-PROBOOK-4540S-HACKINTOSH#12]
Can’t turn down brightness with keyboard button or in macOS. Everything else working great

**Comment by cciRRus** [comment]
Yes, brightness control does not work: keyboard buttons nor Display Preferences, no way to adjust brightness.
I was told that the reason is because the display is detected as an **external display**, instead of a laptop LCD display. We need to somehow change that.

From the [screenshot](https://github.com/nguyenhung9x2020/HP-PROBOOK-4540S-HACKINTOSH/blob/11.2.2/screenshoot/displays.png) provided in this project, we can see that the first display, labeled "Built-in Display", is represented by a picture of an external monitor.

**Comment by jball-98** [comment]
brightness is working now. Everything works except connecting bluetooth devices for some reason

**Comment by cciRRus** [comment]
I discovered the fix: replace **SSDT-PLNF.aml** with RehabMan's version. Get it from:

[https://bitbucket.org/RehabMan/applebacklightfixup/downloads/](https://bitbucket.org/RehabMan/applebacklightfixup/downloads/)

**RehabMan-BacklightFixup-2018-1013.zip** [dated 2018-10-13]

While the zip file contains the kext, it is NOT REQUIRED.

After using this SSDT-PLNF.aml, at Display preferences, it correctly shows the display as a laptop LCD.

Hope this helps!

**Comment by jball-98** [comment]
Didn't have to do that when i installed big sur. worked straight away after installling

does your bluetooth work? i have the same card in the specs on here (BCM94352HMB)

**Comment by cciRRus** [comment]
I started with 11.3.1 and the EFI.zip here. Everything works except the brightness control. I put in RehabMan's SSDT-PNLF.aml, brightness control works. To be absolutely sure, I replaced it with the original SSDT-PNLF.aml in the EFI.zip here, or the official one from OpenCore, and then brightness control fails. So I conclude that the fix is to use RehabMan's SSDT-PNLF.aml.

Sorry, I don't have that wireless module; mine does not support Bluetooth.
